### PR TITLE
Support new block kit builder

### PR DESCRIPTION
**Pull request by barlock**
**Related PRs**  
This PR is not dependent on any other PR

**What does this PR do?**  
It started as an attempt to make the image rendering more stable and fast, it ended up being a complete rewrite b/c block kit builder rolled a new release as I was writing it. 😭 It's also faster and stabler hopefully.

**Description of Changes**  
Use a single page to enter json vs query strings. Also adds a few new features. will comment

**What gif most accurately describes how I feel towards this PR?**

### CODE REVIEW

File: packages/blocks-to-image/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@slack-wrench/blocks-to-image",
-  "version": "1.2.0",
+  "version": "2.0.0",
```
Comment (barlock) [inline packages/blocks-to-image/package.json:3]: 🤷

File: packages/blocks-to-image/src/index.ts [hunk 2]
```diff
@@ -7,11 +8,21 @@ import puppeteer, {
...
 } from 'puppeteer';
-import querystring from 'querystring';
 
 type Screenshot = string | Buffer;
 
-type BlockSurfaceModes = 'message' | 'modal' | 'appHome';
+// Order of the checklist
+enum SurfaceModes {
+  'message',
+  'attachments',
```
Comment (barlock) [inline packages/blocks-to-image/src/index.ts:17]: They added a new one 🚀

File: packages/blocks-to-image/src/index.ts [hunk 3]
```diff
@@ -71,6 +84,32 @@ export default class BlockKitRenderer {
...
+        'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36',
+      );
+      // Keep timestamps the same
+      await this.page.emulateTimezone('GMT');
+      this.page.setDefaultNavigationTimeout(60 * 1000);
+
+      await this.page.goto(`https://api.slack.com/tools/block-kit-builder`);
+      // Render modals as full length
+      await this.page.addStyleTag({
+        content: '.p-bkb_preview_modal { max-height: 100% }',
```
Comment (barlock) [inline packages/blocks-to-image/src/index.ts:106]: You're welcome

File: packages/blocks-to-image/src/index.ts [hunk 4]
```diff
@@ -81,44 +120,200 @@ export default class BlockKitRenderer {
...
+    `);
+
+    let identicalCount = 0;
+    let renderContents = '';
+
+    // Wait for the entered text to have rendered
+    await retry(
+      async () => {
+        // Check for loading or warning icons, multiple of these can happen,
+        // hence it's position in the loop
```
Comment (tteltrab) [inline packages/blocks-to-image/src/index.ts:224]:
```suggestion
        // hence its position in the loop
```

File: packages/blocks-to-image/src/index.ts [hunk 5]
```diff
@@ -81,44 +120,200 @@ export default class BlockKitRenderer {
...
+
+        if (newContents === renderContents) {
+          identicalCount += 1;
+        } else {
+          identicalCount = 0;
+        }
+
+        renderContents = newContents;
+
+        // Magic number
```
Comment (tteltrab) [inline packages/blocks-to-image/src/index.ts:241]: lol, but where did it come from?

File: packages/blocks-to-image/src/index.ts [hunk 6]
```diff
@@ -81,44 +120,200 @@ export default class BlockKitRenderer {
...
+  }
+
+  private async selectMode(mode: SurfaceModes): Promise<void> {
+    const page = await this.getBuilderPage();
+
+    await page.click('#block-kit-builder-surface-select_button');
+    const switchElement = await page.waitForSelector(
+      `#block-kit-builder-surface-select_option_${mode}`,
+    );
+    await switchElement.click();
```
Comment (tteltrab) [inline packages/blocks-to-image/src/index.ts:261]: 👍

**Comment by tteltrab** [comment]
LGTM. Some small issues but we can fix/update some of this in a patch version. Worth shipping asap

**Pull request merged**